### ISSUE TITLE
(inversify-express-utils) Add the ability to support using an existing express application instance

### ISSUE THREAD

**Issue by ezra-quemuel** [inversify/InversifyJS#509]
I'd like the ability to provide InversifyExpressServer an existing instance of an express application. 

## Expected Behavior/Possible Solution
To be able to pass in an existing express application in the constructor in this fashion:
```javascript
const app = express();
const server = new InversifyExpressServer(container, null, null, app);
```
Or something along those lines...
I'd be happy to submit a PR if this is an acceptable solution.

## Context
Currently I am using a library that manages the express server (sets up some boilerplate middleware/routes and then calls listen() on it) so I only have access to the express server through a configuration method hook exposed on the provided class. This is what it looks like

```javascript
//Base server handles calling .listen() and adding boilerplate routes/middleware
class HTTPServer extends BaseServer {
  setupServer() {
    // Do stuff with this.app
    this.app.use(...);
  }
}
```

## Your Environment

Node.js: 6.10.0
inversify: 3.10

**Comment by lholznagel** [comment]
From a quick look, this could be possible.
Just keep in mind that the decorators are still needed.

**Comment by remojansen** [comment]
Sorry I was unable to release yesterday. I will try this evening again. I'm finishing a new feature int he core lib...

**Comment by remojansen** [comment]
I have released inversify-express-utils@3.3.0 & inversify@3.2.0 🎉  I'm closing this issue.
